### PR TITLE
Adding Poll model

### LINKED ISSUE #28: Adding the Poll model
- Creating the Poll table
- Testing the Poll table

### PR DESCRIPTION
**Pull request by OmerCS8**
- Adding `Poll` model that inherits from Post
- Adding test to `Poll`
fix #28

### CODE REVIEW

File: posts/tests.py [hunk 1]
```diff
@@ -1,3 +1,88 @@
...
+    # Check if Poll deletion delete only Poll from database.
+    def test_delete_poll(self, persist_Poll):
+        assert persist_Poll in Poll.objects.all()
+        assert persist_Poll.author in User.objects.all()
+        persist_Poll.delete()
+        assert persist_Poll not in Poll.objects.all()
+        assert persist_Poll.author in User.objects.all()
+
+    # Check if Poll's author deletion delete both the author and the poll.
+    def test_delete_polls_aouthor(self, persist_Poll):
```
Comment (AlonShakaroff) [inline posts/tests.py:73]: There's a type in the method name, its suppose to be author.

Comment (OmerCS8) [reply posts/tests.py]: Fixed

**Review by MatanP12 — APPROVED** [review]
LGTM

File: posts/tests.py [hunk 2]
```diff
@@ -83,3 +95,51 @@ def test_resume_deletion_after_author_deletion(self, persist_resume):
...
+        assert new_poll.author.username == USERNAME
+        assert new_poll.description == DESCRIPTION
+
+    # Check if the Poll is saved in the database.
+    def test_persist_poll(self, persist_Poll):
+        assert persist_Poll in Poll.objects.all()
+
+    # Check if the Poll can be referred as post.
+    def test_inheritance_of_poll_from_post(self, persist_Poll):
+        assert isinstance(persist_Poll, Post)
```
Comment (AlonShakaroff) [inline posts/tests.py:116]: Same as Nati told me, you can use:
`assert Post.objects.filter(poll=persist_poll).exists()`
Instead of isinstance()

Comment (OmerCS8) [reply posts/tests.py]: Changed it 👍

**Review by AlonShakaroff — APPROVED** [review]
LGTM

**Review by MatanP12 — DISMISSED** [review]
LGTM

**Review by TomerNewmanPrograms — APPROVED** [review]

**Review by yulisuliman — DISMISSED** [review]

File: posts/tests.py [hunk 3]
```diff
@@ -226,3 +236,52 @@ def test_rating_invalid_args(
     # Check if the Rating average function works properly
     def test_rating_avg_function(self, persist_rating):
         assert persist_rating.get_rating_average() == 5
+
+
+# -------------------------------------------- Poll tests --------------------------------------------
+
+
```
Comment (natifridman) [inline posts/tests.py:243]: Please follow pep8 for blank lines

File: posts/tests.py [hunk 4]
```diff
@@ -226,3 +236,52 @@ def test_rating_invalid_args(
...
+@pytest.mark.django_db()
+class TestPoll:
+
```
Comment (natifridman) [inline posts/tests.py:246]: Please follow pep8 for blank lines

File: posts/tests.py [hunk 5]
```diff
@@ -226,3 +236,52 @@ def test_rating_invalid_args(
...
+        assert new_poll.description == DESCRIPTION
+
+    # Check if the Poll is saved in the database.
+    def test_persist_poll(self, persist_poll):
+        assert persist_poll in Poll.objects.all()
+
+    # Check if the Poll can be referred as post.
+    def test_inheritance_of_poll_from_post(self, persist_poll):
+        assert Post.objects.filter(poll=persist_poll).exists()
+        post = Post.objects.all().first()
```
Comment (natifridman) [inline posts/tests.py:259]: Is it always going to be the first? I think we can get it with the filter

**Review by yulisuliman — APPROVED** [review]

**Review by MatanP12 — COMMENTED** [review]
LGTM

File: posts/tests.py [hunk 6]
```diff
@@ -226,3 +236,52 @@ def test_rating_invalid_args(
...
+        assert persist_poll not in Poll.objects.all()
+        assert persist_poll.author not in User.objects.all()
+
+    # verify that it is impossible to create a Poll with invalid parameters.
+    @pytest.mark.parametrize("user, description, expected_error", [(
+        USERNAME,
+        DESCRIPTION,
+        TypeError),
+        (new_user, 6, TypeError), (EMAIL, DESCRIPTION, TypeError)])
+    def test_poll_invalid_args(self, user, description, expected_error):
```
Comment (abiton1) [inline posts/tests.py:282]: You don't need to test all these variations of values \
The test is for not been able to create a `Poll` object with `user` and `description` \
the values are not relevant here, one test should cover it.

Comment (OmerCS8) [reply posts/tests.py:282]: Yes, as you said I meant to test it with `Poll(author=user, description=description)`, my bad.
I'll fix it 👍

Comment (OmerCS8) [reply posts/tests.py:281]: fixed

**Review by TomerNewmanPrograms — DISMISSED** [review]

**Review by yulisuliman — APPROVED** [review]

**Comment by natifridman** [comment]
Please fix the conflicts

**Comment by OmerCS8** [comment]
> Please fix the conflicts
Fixed all conflicts 👍

**Review by MatanP12 — APPROVED** [review]
LGTM!

**Review by TomerNewmanPrograms — APPROVED** [review]
lgtm

**Review by natifridman — APPROVED** [review]

**Pull request merged**